ibis/expr/operations/reductions.py

```diff
@@ -376,6 +376,15 @@ class ArrayCollect(Filterable, Reduction):
     arg: Column
     order_by: VarTuple[SortKey] = ()
     include_null: bool = False
+    distinct: bool = False
+
+    def __init__(self, arg, order_by, distinct, **kwargs):
+        if distinct and order_by and [arg] != [key.expr for key in order_by]:
+            raise ValidationError(
+                "`collect` with `order_by` and `distinct=True` and may only "
+                "order by the collected column"
```

Comment:
No backend supports this, so I opted to error in our IR rather than at runtime.

Reply:
Seems fine, but generally we should avoid overriding `__init__` in most cases. For more complicated validations like this one, overriding `__init__` is often the only way to reasonably spell the conditon.

Reply:
Yeah, that's why I did it here.